agent/engine/docker_task_engine.go

```diff
@@ -554,6 +577,20 @@ func (engine *DockerTaskEngine) pullAndUpdateContainerReference(task *api.Task,
 		return DockerContainerMetadata{Error: TaskStoppedBeforePullBeginError{task.Arn}}
 	}
 
+	// Set the credentials for pull from ecr if necessary
+	if container.ShouldPullWithExecutionRole() {
+		executionCredentials, ok := engine.credentialsManager.GetTaskCredentials(task.GetExecutionCredentialsID())
+		if !ok {
+			seelog.Infof("Acquiring ecr credentials from credential manager failed, container: %s, task: %s", container.String(), task.String())
+			return DockerContainerMetadata{Error: CannotPullECRContainerError{errors.New("engine ecr credentials acquisition: not found in credential manager")}}
+		}
+
+		iamCredentials := executionCredentials.GetIAMRoleCredentials()
+		container.SetRegistryAuthCredentials(iamCredentials)
```

Comment:
This method returns an error. You should handle that. Or, split `SetRegistryAuthCredentials` into `Validate...` and `Set...` and handle `Validate...` errors. Also needs unit tests for when an error occurs here.

Reply:
The validate is covered by the `ShouldPullWithExecutionRole`, I will remove the validate for this function